thorlcr/activities/hashdistrib/thhashdistribslave.cpp

```diff
@@ -2198,7 +2198,12 @@ class CHDRproportional: public CSimpleInterface, implements IHash
                 MemoryAttr ma;
                 activity->startInput(in);
                 if (activity->getOptBool(THOROPT_COMPRESS_SPILLS, true))
+                {
                     rwFlags |= rw_compress;
+                    StringBuffer compType;
+                    activity->getOpt(THOROPT_COMPRESS_SPILL_TYPE, compType);
+                    setCompFlag(compType, rwFlags);
+                }
```

Comment:
Looks like these lines are needed in a few places, would be good to common up.
Hash distribute (CDistributorBase) fetches a ICompressHandler based on activity/workunit/global properties. That code could be commoned up in the activity base class for all to use.